### ISSUE TITLE
Improve [for efficiency] the flow to add a new GUI object from scratch.

### ISSUE THREAD

**Issue by LuniMoon** [4ian/GDevelop#5938]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/4ian/GDevelop/issues)

### Describe the bug

> Based on Keith_1357's request [on the forum](https://forum.gdevelop.io/t/after-you-add-user-interface-objects-it-should-show-the-edit-screen/52030).

### Today's System:
For the flow "New object from Scratch" when it comes to GUI elements
<img width="505" alt="Screenshot 2023-11-22 at 10 44 24" src="https://github.com/4ian/GDevelop/assets/103995399/32751ef8-58c7-42c0-a575-6f012e9a11d7">

The system asks the user for their selected ressource
<img width="498" alt="Screenshot 2023-11-22 at 10 53 32" src="https://github.com/4ian/GDevelop/assets/103995399/4f880c28-5433-477c-870b-30552d845cf3">

And after its selection, it sends the user to the Scene (closes all the modals).

-------

### Suggested Flow:
(Increase efficiency)

- The user selects its GUI element
- The system opens the Object Properties
<img width="485" alt="Screenshot 2023-11-22 at 11 05 14" src="https://github.com/4ian/GDevelop/assets/103995399/5f8dc39b-1d91-45b7-9d83-d2fc1bed0adf">

to reflect the same flow that the system does when adding other type of object, and to increase efficiency.

### Steps to reproduce

1. Add a new object from scratch 
2. Select a GUI element (button, slider, bar)
3. The system will take you to the Scene screen.

### GDevelop platform

Desktop, Web, Mobile

### GDevelop version

5.3.181

### Platform info

<details>

*OS (e.g. Windows, Linux, macOS, Android, iOS)*
>

*OS Version (e.g. Windows 10, macOS 10.15)*
>

*Browser(For Web) (e.g. Chrome, Firefox, Safari)*
>

*Device(For Mobile) (e.g. iPhone 12, Samsung Galaxy S21)*
>

</details>

**Comment by 4ian** [comment]
I'm not sure we want to go in this direction.
These objects are "custom objects", which can be sensibly longer to configure than traditional objects because they are more complex, especially if you want to get a good looking result.

We have a button for starting from 0, it's called "Skip and create from scratch" - I don't have numbers but I assume usually it's just faster to start with an existing button (and we can add more when some are made by the community).

**Comment by LuniMoon** [comment]
I agree. Custom objects require some fine work and tuning and the V0 UI screen is cognitive heavy.
I do have memories of having this "the custom object parameters are complex" conversation when we did the "skip and create from scratch" button for V0.

However, the creation flow is not as the one that a user would have found in other elements.
We could argue that it is to protect the user from a heavy UI screen, but I think that not respecting the patterns hurts system reliability: A user will "muscle memorise" that X action is located on W corner which button is a certain colour. While using software, we rely and trust that these learned patterns because they repeat across the UI.

However on GDevelop these patterns could be irregular depending on the flow.
On this screen, the Primary button - which is often "Accept" on these flows - is "Skip and create from scratch". Also, after clicking a custom object; the user cannot "cancel" because the system selects the object (without them clicking "Accept"), closes the modal, and automatically goes to the Scene.
This is not the pattern that is present in the flow for other object types. This issue changes system reliability requiring time to carefully read the buttons (impacting efficiency).

The requested change is to respect the UI pattern across the platform. I can eventually take a"how to improve button custom object usability" so the apparition of that screen is "less frightening".
As for how this change will impact users now... I don't think it's a dangerous move *because* users have already seen and interacted with the properties screen from other object types. My guess is that they'll know how to navigate it/use it for a custom object too.

The suggested button structure for this custom object dialog could be: Cancel [Secondary], Skip and create from scratch [Secondary], Accept [Primary].

**Comment by D8H** [comment]
> On this screen, the Primary button - which is often "Accept" on these flows - is "Skip and create from scratch".

What do you suggest?
Do you think that users could miss the button and a thumbnail should also be added for an empty object?

> Also, after clicking a custom object; the user cannot "cancel" because the system selects the object (without them clicking "Accept"), closes the modal, and automatically goes to the Scene.

Even if the object creation was cancelled, the resources would remain in the project. I'm not sure there is an easy way around this.

> The requested change is to respect the UI pattern across the platform.

I think that the current behavior doesn't break the UI logic:
- the user choose a ready to use configuration, I bet they don't need to change the configuration 99% of the time, so they don't need to see an extra dialog just to close it
- the user choose to setup the object, the object editor is shown

**Comment by 4ian** [comment]
On the forum issue, there were some alternative suggestions. I think always opening the object editor at the end was one of them.

**Comment by LuniMoon** [comment]
I'll back pedal a little bit because this issue is pointing to a flow-specific UX bug (adding custom UI object types to the project), but this flow is not optimal because it is a consequence of a problem that contains this particular UI interaction. I'll explain:

Said "zoomed-out" problem is the fact that in certain dialogs (modals), the buttons do not work as "promised" by the copy they display: a "Cancel" button does not cancel the action (an object was created on the object list), a "Close" button closes a modal but it isn't clear if the changes made on that level are saved... and so on -[see related forum issue here](https://forum.gdevelop.io/t/github-issue-5938-when-cancel-is-clicked-after-adding-an-object-the-object-shouldnt-be-created/52031)-.
These UI inconsistencies impact the efficiency of "Adding a GUI object from scratch", _and_ other flows -[see Tristan's feedback on core discord](https://discord.com/channels/258623956158906368/926423098574663680/1181792685401591908)-. The output of what a "cancel" button is, can vary from modal to modal, and that impacts: trust in the system (the outcome is hard to predict), and efficiency (the user has to correct what the product did against their expectations).

Now, back to this issue: There is a modal step in which the "Accept" or "Apply" button that is present in all dialogs, is not there and it is not following the same steps as equivalent flows...  I think this is a consequence of the UX issue that dialogs aren't very consistent (which I think is the root of this, and similar frictions).

A bigger action to execute is to standardise _all_ modal buttons (copy, and what they do) to improve efficiency and product trust.
A smaller scope (this issue for this flow) would be:
On the "new object from scratch" modal for UI components:
- The user clicks on a selected object and the system does nothing (this would allow the user to change their choice in case it is not their final).
- To select an object and add it to their object list, the user has to click an "Accept" button (currently inexistent).

These changes leave us with the following buttons:
- Close (secondary) -> Currently there -_tho I wonder if "Cancel" isn't a better term for this so it's clear for the user that the system won't process any change on that dialog_-.
- Skip and create from scratch (secondary) -> Existent, but added as primary
- Accept (primary) -> Not existent yet.

As for opening the object parameters just after (just as adding a sprite does)... I think [the user is right](https://forum.gdevelop.io/t/github-issue-5938-after-you-add-user-interface-objects-it-should-show-the-edit-screen/52030/5?u=luni): we should open it for consistency with other menus. _Tho_, there is a real concern of not wanting to show that screen because it could appear cognitive heavy, or because it won't be needed 99% of the time by other types of users. To make a decision on this, I would check any usability data (if we have it, and it not we might want to consider it) to have an idea on what percentage is opening their custom object dialogs to personalise their values. I do agree with Davy and think that it won't be more than 60%... for now, [asking the user](https://forum.gdevelop.io/t/github-issue-5938-after-you-add-user-interface-objects-it-should-show-the-edit-screen/52030/7?u=luni) is the only way I can think of to acquire that info.
